src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -839,11 +854,40 @@ public void SetAbi (string abi)
 			int code = 1;
 			if (!string.IsNullOrEmpty (VersionCode)) {
 				code = Convert.ToInt32 (VersionCode);
-				if (code > 0xffff || code < 0)
-					throw new ArgumentOutOfRangeException ("VersionCode", "VersionCode is outside 0, 65535 interval");
+				if (code > maxVersionCode || code < 0)
+					throw new ArgumentOutOfRangeException ("VersionCode", $"VersionCode is outside 0, {maxVersionCode} interval");
 			}
 			code |= GetAbiCode (abi) << 16;
 			VersionCode = code.ToString ();
 		}
+
+		public void CalculateVersionCode (string currentAbi, string versionCodePattern, string versionCodeProperties)
+		{
+			var regex = new Regex ("\\{(?<key>([A-Za-z]*)):?0*[\\}]");
```

Comment:
Do we really want `(?<key>([A-Za-z]*))`? That would match *no* characters, allowing a match of `{}`. Is that sensible?

Additionally, does the `0*` make sense? That looks like it should be *any* format string, so what would be wrong with e.g. `{versionCode:x8}`?